### PR TITLE
feat: add validation, basic error handling & documentation for api

### PR DESCRIPTION
**Pull request by ritikrajdev**
NOTE: this is an extension of the work done in #9 and #5 .
Changes done :-
* Addition of overall documentation of the api using jsdoc ad swagger ui express
* creation of middleware to handle errors
* joi validation for all endpoints

### CODE REVIEW

File: src/controllers/poNotes.controller.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  getPONotesByQuickFilter,
+  createValidPONote,
+  updatePONoteById,
+  hardDeletePONoteById,
+  softDeletePONoteById
+} = require('../services/poNoteServices');
+
+const listPONotes = async (req, res) => {
+  try {
+
```
Comment (sugumar-mani) [inline src/controllers/poNotes.controller.js:20]: remove unused space

File: src/controllers/poNotes.controller.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  catch (er) {
+    res.status(404).json({
+      message: er.message
+    });
+  }
+};
+
+const createPONote = async (req, res) => {
+  try {
+
```
Comment (sugumar-mani) [inline src/controllers/poNotes.controller.js:52]: remove unused space

File: src/controllers/poNotes.controller.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    // optional
+    const validDueDate = req.body?.duedate ?
```
Comment (sugumar-mani) [inline src/controllers/poNotes.controller.js:54]: Does this mean the due date is mandatory? We should keep it optional

File: src/utils/prismaUtils.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+const statusEnum = {
+  COMPLETED: 'COMPLETED',
+  PENDING: 'PENDING',
+  NONE: 'NONE'
+};
+
+const typeEnum = {
+  ACTION_ITEM: 'ACTION_ITEM',
+  KEY_DECISION: 'KEY_DECISION',
+  AGEND_ITEM: 'AGEND_ITEM'
```
Comment (sugumar-mani) [inline src/utils/prismaUtils.js:40]: spelling mistake

Comment (ritikrajdev) [reply src/utils/prismaUtils.js:40]: Hi Sugumar, this has been removed in latest pull request as we are using joi validation to make validation easier [`77ed512` (#11)](https://github.com/tech-university-india/abd-backend/pull/11/commits/77ed512aa8b4356f8c8d4e143fb2dfa07e04f26f).

Thanks

**Pull request closed without merging**